test-adapter/src/main/scala/org/scalajs/testadapter/TestAdapter.scala

```diff
@@ -176,18 +170,24 @@ final class TestAdapter(jsEnv: ComJSEnv, jsFiles: Seq[VirtualJSFile],
 }
 
 object TestAdapter {
+  sealed abstract class ModuleIdentifier
```

Comment:
Maybe add a comment that `sealed` is not part of the public API?

Reply:
Done. I also used your trick for the other PR with a non-sealed class whose constructor is `private`, so that is actually not `sealed`. We only have to match on it within our own code, and I would hope that we will always have a least one test for each module kind ;)